contracts/implementations/streamr/VehicleStream.sol

```diff
@@ -0,0 +1,102 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.13;
+
+import "../../interfaces/INFT.sol";
+import "../../interfaces/IStreamRegistry.sol";
+import "../../libraries/nodes/VehicleStorage.sol";
+import "../../libraries/streamr/StreamrManagerStorage.sol";
+import "../../libraries/streamr/VehicleStreamStorage.sol";
+
+import "../../shared/Errors.sol" as Errors;
+
+import "@solidstate/contracts/access/access_control/AccessControlInternal.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+/// TODO Documentation
+contract VehicleStream is AccessControlInternal {
+    string constant DIMO_STREAM_ENS = "streams.dimo.eth";
+    string constant DIMO_STREAM_ENS_VEHICLE = "streams.dimo.eth/vehicle/";
+
+    // TODO Documentation
+    function createStream(uint256 vehicleId) external {
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        IStreamRegistry streamRegistry = IStreamRegistry(
+            StreamrManagerStorage.getStorage().streamrRegistry
+        );
+
+        try INFT(vehicleIdProxyAddress).ownerOf(vehicleId) returns (
+            address vehicleOwner
+        ) {
+            if (vehicleOwner != msg.sender) {
+                revert Errors.Unauthorized(msg.sender);
+            }
+        } catch {
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleId);
+        }
+
+        string memory streamPath = string(
+            abi.encodePacked("/vehicle/", Strings.toString(vehicleId))
+        );
+        streamRegistry.createStreamWithENS(DIMO_STREAM_ENS, streamPath, "{}");
+    }
+
+    // TODO Documentation
+    function subscribe(
```

Comment:
This doesn't seem safe. What's the plan for this function?

Reply:
It is to allow a user to subscribe to someone's stream. Still missing a couple of checks